packages/flutter_tools/lib/src/devtools_launcher.dart

```diff
@@ -31,41 +32,44 @@ class DevtoolsServerLauncher extends DevtoolsLauncher {
   final Logger _logger;
 
   io.Process _devToolsProcess;
-  Uri _devToolsUri;
 
   static final RegExp _serveDevToolsPattern =
       RegExp(r'Serving DevTools at ((http|//)[a-zA-Z0-9:/=_\-\.\[\]]+)');
 
   @override
   Future<void> launch(Uri vmServiceUri, {bool openInBrowser = false}) async {
-    if (_devToolsProcess != null && _devToolsUri != null) {
+    if (_devToolsProcess != null && devToolsUri != null) {
       // DevTools is already running.
       if (openInBrowser) {
-        await Chrome.start(<String>[_devToolsUri.toString()]);
+        await Chrome.start(<String>[devToolsUri.toString()]);
       }
       return;
     }
 
-    final Status status = _logger.startProgress(
-      'Activating Dart DevTools...',
-    );
+    bool offline = false;
     try {
-      // TODO(kenz): https://github.com/dart-lang/pub/issues/2791 - calling `pub
-      // global activate` adds ~ 4.5 seconds of latency.
-      final io.ProcessResult _devToolsActivateProcess = await _processManager.run(<String>[
-        _pubExecutable,
-        'global',
-        'activate',
-        'devtools'
-      ]);
-      if (_devToolsActivateProcess.exitCode != 0) {
-        status.cancel();
-        _logger.printError('Error running `pub global activate '
-            'devtools`:\n${_devToolsActivateProcess.stderr}');
+      await http.head('https://pub.dev');
```

Comment:
Make sure to include a check for https://flutter.dev/community/china#configuring-flutter-to-use-a-mirror-site , since otherwise this will fail in China.

Reply:
done

Reply:
`https://pub.dev` won't be accessible though, so one of these will always fail. Instead you should look up if there is a pub url override in the environment variable (the user could have some arbitrary pub mirror configured) and use that if present, falling back to pub.dev